test/object-store/realm.cpp

```diff
@@ -2693,6 +2693,28 @@ TEST_CASE("SharedRealm: async writes") {
         REQUIRE(done == true);
     }
 
+    SECTION("open new realm with diffent schema while async transaction is in progress") {
+        realm->async_begin_transaction([&] {
+            done = true;
+        });
+
+        config.schema = Schema{
+            {"hotel",
+             {
+                 {"price", PropertyType::Int},
+                 {"city", PropertyType::String},
+             }},
+        };
+
+        try {
+            auto r2 = Realm::get_shared_realm(config);
+            REQUIRE(!r2);
```

Comment:
shouldn't `done` be false here? (we should assert it)

Reply:
We should never get here